relay-redis/src/real.rs

```diff
@@ -144,68 +231,127 @@ pub struct RedisPool {
 }
 
 impl RedisPool {
-    /// Creates a `RedisPool` in cluster configuration.
+    /// Creates a [`RedisPool`] in cluster configuration.
     pub fn cluster<'a>(
         servers: impl IntoIterator<Item = &'a str>,
         opts: RedisConfigOptions,
     ) -> Result<Self, RedisError> {
-        let pool = Pool::builder()
-            .max_size(opts.max_connections)
-            .min_idle(opts.min_idle)
-            .test_on_check_out(false)
-            .max_lifetime(Some(Duration::from_secs(opts.max_lifetime)))
-            .idle_timeout(Some(Duration::from_secs(opts.idle_timeout)))
-            .connection_timeout(Duration::from_secs(opts.connection_timeout))
+        let pool = Self::base_pool_builder(&opts)
             .build(redis::cluster::ClusterClient::new(servers).map_err(RedisError::Redis)?)
             .map_err(RedisError::Pool)?;
 
         let inner = RedisPoolInner::Cluster(pool);
         Ok(RedisPool { opts, inner })
     }
 
-    /// Creates a `RedisPool` in single-node configuration.
+    /// Creates a [`RedisPool`] in multi write configuration.
+    pub fn multi_write<'a>(
+        primary: &str,
+        secondaries: impl IntoIterator<Item = &'a str>,
+        opts: RedisConfigOptions,
+    ) -> Result<Self, RedisError> {
+        let primary_pool = RedisPoolInner::Single(Self::client_pool(primary, &opts)?);
+        let secondary_pools = secondaries
+            .into_iter()
+            .map(|s| Self::client_pool(s, &opts))
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .map(RedisPoolInner::Single)
+            .collect();
```

Comment:
You could write this as 
```suggestion
            .map(|s| Self::client_pool(s, &opts).map(RedisPoolInner::Single))
            .collect::<Result<Vec<_>, _>>()?;

```
to save the inner `collect`.

Reply:
Oh yeah good idea! Thanks!